### PR TITLE
reposition stripe payment complete reporting and add metric to track unhandled stripe failures

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
This pull request repositions payment complete reporting in Stripe.controller so that it fires at the very last moment before user is redirected to thank-you page.

This is because there have been a number of false alarms as a result of payment being successful, but user not being shown the thank-you page. This calls the use of thank-you page as a reliable indicator of a successful payment into question and I have changed the librato metrics to instead alert on the Stripe Payment Success metric insted.
To minimise the chance of this alert not being triggered in the event of an unsuccessful payment, I have moved it into the redirect logic called at the very end of the "pay" function.

This pull request also adds a new metric to capture any Stripe payment failures that are not a direct result of Stripe responding that it does not accept thee payments. The idea of this metric is to capture edge cases such as timeouts.

Have you gone through the contributions flow for all test variants ?

### CODE REVIEW

File: app/monitoring/CloudWatchMetrics.scala [hunk 1]
```diff
@@ -67,6 +67,10 @@ class CloudWatchMetrics(cloudWatchClient: AmazonCloudWatchAsync) extends TagAwar
     put("payment-failure", paymentProvider, platform)
   }
 
+  def logUnhandledPaymentFailure(paymentProvider: PaymentProvider, platform: String): Unit = {
+    put("payment-failure", paymentProvider, platform)
```
Comment (desbo) [inline app/monitoring/CloudWatchMetrics.scala]: Should this be called `unhandled-payment-failure` ?

Comment (michaelwmcnamara) [reply app/monitoring/CloudWatchMetrics.scala]: omg yes it should!!!

Comment (michaelwmcnamara) [reply app/monitoring/CloudWatchMetrics.scala]: Latest commit fixes this.

File: app/controllers/StripeController.scala [hunk 2]
```diff
@@ -130,8 +132,6 @@ class StripeController(paymentServices: PaymentServices, stripeConfig: Config, c
     }
 
     createCharge.map { charge =>
-      info(s"Stripe payment successful for request id: ${request.id}, from platform ${request.platform}")
```
Comment (desbo) [inline app/controllers/StripeController.scala]: I think this change should be reverted, since it doesn't look like moving this into the `thankYouUri` call will make any difference. In both cases, the calls would be executed before the user gets the redirect response.

It'd be good to keep the `thankYouUri` function as something that purely generates a URI without any unrelated side effects - what about a new function to take care of the logging stuff?

Comment (michaelwmcnamara) [reply app/controllers/StripeController.scala]: These are good points! - this is fixed in latest commit.

**Review by desbo — APPROVED** [review]
